src/main/java/org/interledger/cryptoconditions/FulfillmentFactory.java

```diff
@@ -0,0 +1,53 @@
+package org.interledger.cryptoconditions;
+
+import java.lang.reflect.Constructor;
+
+import org.interledger.cryptoconditions.encoding.Base64Url;
+
+public class FulfillmentFactory {
+	private static final String FULFILLMENT_REGEX = "^cf:([1-9a-f][0-9a-f]{0,3}|0):[a-zA-Z0-9_-]*$";
+	private static final java.util.regex.Pattern p = java.util.regex.Pattern.compile(FULFILLMENT_REGEX);
+	
+	public static Fulfillment getFulfillmentFromURI(String URI){
```

Comment:
We should use a FulfillmentReader and then have this as a convenience function that wraps that.